### PR TITLE
Added newer versions of py-torch-sparse and py-torch-geometric

### PR DESCRIPTION
**Pull request by brandon-feder**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py [hunk 1]
```diff
@@ -20,7 +20,9 @@ class PyTorchGeometric(PythonPackage):
 
     homepage = "https://github.com/rusty1s/pytorch_geometric"
     url      = "https://github.com/rusty1s/pytorch_geometric/archive/1.6.0.tar.gz"
-
+   
+    version('2.0.4', sha256='61b00a8c1d76455c65055645899492b37642c1bfd65a3b9beb8fe2a906d11bff')
+    version('1.7.2', sha256='33663e84c97ba9a35bf0c15a63c1c7d9b2aec7a2178e92e39deb6712ce2fbb27')
     version('1.6.3', sha256='1907c0f5eb7ab8f6f79a7e43703579b39b514501dd956ed0fc3d33210bcbb4c2')
     version('1.6.0', sha256='7d5231cdcc2ebd4444f406cbf1537eb49bf90ab6f446eaf1b7af5cdbe105f3c9')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-geometric/package.py:28]: Can you update the list of dependencies?
```
install_requires = [                                                                     
    'tqdm',                                                                              
    'numpy',                                                                             
    'scipy',                                                                             
    'pandas',                                                                            
    'jinja2',                                                                            
    'requests',                                                                          
    'pyparsing',                                                                         
    'scikit-learn',                                                                      
]       
    python_requires='>=3.7',                                                             
```
Everything else is optional in this latest release.

File: var/spack/repos/builtin/packages/py-torch-sparse/package.py [hunk 2]
```diff
@@ -12,7 +12,8 @@ class PyTorchSparse(PythonPackage):
 
     homepage = "https://github.com/rusty1s/pytorch_sparse/"
     url      = "https://github.com/rusty1s/pytorch_sparse/archive/0.6.7.tar.gz"
-
+    
+    version('0.6.14', sha256='b1ad6b78410d7fddabe07799c51d539badb6fd962ba3616c1eeb23bfe7bd5c59')
     version('0.6.8', sha256='98f7ff1f0f9cd5031bc81c70c11970c3864545ae33677025a6efd2466a97e6f9')
     version('0.6.7', sha256='0d038a1502548692972a085cd0496460b5d2050bb7328427add990f081d6c44d')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-sparse/package.py:19]: Need to update dependency list:
```
install_requires = [                                                                     
    'scipy',                                                                             
]    
    python_requires='>=3.7',                                                             
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html for instructions on finding and adding Python dependencies

**Comment by brandon-feder** [comment]
Doesn't both`py-torch` and `py-torch-geometric` depend on more than just what you listed? 

When you write 
```
install_requires = [                                                                     
    'scipy',                                                                             
]   
```
Do you want me to remove everything but `depends_on('py-scipy', type=('build', 'run'))`?

I apologize for my misunderstanding.

**Comment by adamjstewart** [comment]
In the case of `py-torch-sparse`, it looks like many of the package dependencies aren't explicitly listed but still required. I think `py-pytest-runner` isn't required (at least in the latest version) but the other dependencies are still required. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html for docs on how to determine which dependencies are required.

**Comment by tgamblin** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot hello

**Comment by scottwittenburg** [comment]
@spackbot help

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by adamjstewart** [comment]
@tgamblin @scottwittenburg should we create a new fake PR to debug spackbot instead of bogging down this real PR with comments?

**Comment by scottwittenburg** [comment]
> @tgamblin @scottwittenburg should we create a new fake PR to debug spackbot instead of bogging down this real PR with comments?

Sorry about this, maybe I can delete these comments?  I did create a fake [PR](https://github.com/spack/spack/pull/31757) to test it, and it's working there.

**Comment by tgamblin** [comment]
@adamjstewart probably :)

I was hoping this would work and the style would be magically fixed.  Any chance you could open a PR for @scottwittenburg to test this on?  We need one not authored by him, because the issue is permissions.  e.g., it works here: https://github.com/spack/spack/pull/31757#issuecomment-1196037483 but @scottwittenburg owns that PR.

**Comment by scottwittenburg** [comment]
@spackbot fix style

**Comment by tgamblin** [comment]
Ok sorry for the noise -- but now it appears that spackbot is working.  So if you want to fix style on your PR (and it's fixable) you'll be able to ask it to do so.  This PR was particularly interesting for testing as @brandon-feder is a first-time contributor and we had a harder time getting spackbot to work for this PR.  Sorry for the noise!

**Comment by adamjstewart** [comment]
But isn't spackbot supposed to commit the changes? The PR hasn't been updated.

**Comment by scottwittenburg** [comment]
> But isn't spackbot supposed to commit the changes

Maybe the message needs to be made clearer.  It does say it cannot fix flake8 or mypy errors, and that it wasn't able to make any further changes on this PR.  But I'm not sure that really stands out.

**Comment by adamjstewart** [comment]
I guess this particular issue will be fixed once we add black, but yeah, it would be good to make it more clear that spackbot failed to fix the failing CI

**Comment by adamjstewart** [comment]
What's the status of this PR? Both packages have since been updated, although these exact versions aren't yet listed. Do you want to try to rebase, or open a new PR, or abandon it?

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**